### ISSUE TITLE
Prevent commented out tests

### ISSUE THREAD

**Issue by sindresorhus** [avajs/eslint-plugin-ava#84]
We should recommend using `test.skip()` instead.

This:

``` js
// test('foo', => {
//  t.pass();
// });
```

Should instead be:

``` js
test.skip('foo', => {
    t.pass();
});
```

**Comment by jamestalmage** [comment]
:+1: 

Interesting proposal.

How should we accomplish it? Try parsing the contents of the comments, or just use regexp?

Trying to parse will have more false negatives, regexp more false positives.

**Comment by jfmengels** [comment]
I found https://github.com/bahmutov/eslint-rules#no-commented-out-code that looks like this. I have no idea if it works well though, but it might be a good lead.

**Comment by jamestalmage** [comment]
It looks like it parses line by line, and warns if any single line is valid JS.

This is a little more difficult to do with parsing because tests will (very likely) cover multiple lines:

```js
test(t => {
  t.pass();
});
```

The first line above isn't valid JS on it's own, you need the whole thing. But if they do this when they comment it out:

```js
// I am skipping this because: reasons.
// 
// test(t => {
//   t.pass();
// });
```

Then parsing the entire contents of the comments creates a syntax error.

I guess we could create a regexp to try and find the potential first line of a test, and then add lines incrementally until we found something that parsed.

Or maybe we could try the [`parse_dammit`](https://github.com/ternjs/acorn#distacorn_loosejs) method from `acorn`.

**Comment by sindresorhus** [comment]
I was thinking regex.

**Comment by jamestalmage** [comment]
> I was thinking regex.

Uh oh: #85 

My concern with regexp is the likelihood of false positives.

**Comment by jamestalmage** [comment]
Acorns loose parser is surprisingly good (at making bad code into a usable AST), it might allow just as many false positives through as a regex would.

**Comment by sindresorhus** [comment]
Yeah, parsing looks like a more solid solution.

**Comment by jamestalmage** [comment]
I think [this](https://github.com/jamestalmage/eslint-plugin-ava/blob/1cd1440086b7106735fc89a41bce1c64341ef662/test/no-commented-tests.js#L25) would be difficult to handle without parsing.

**Comment by jfmengels** [comment]
I found @platinumazure's eslint-plugin-qunit, which has this rule alreadby built-in (https://github.com/platinumazure/eslint-plugin-qunit/blob/master/lib/rules/no-commented-tests.js), and uses a regex to find it. In QUnit's case, there are bounds to be less false positives.

**Comment by revelt** [comment]
I came here looking to check does it exist and if necessary, to propose it :) 

Imho, this rule would be one of the most useful-ones from what I've seen so far in the proposals. It's very difficult to pinpoint commented-out rule when the tests file is long.
